.gitignore

```diff
@@ -16,3 +16,4 @@ cabal.sandbox.config
 *.hp
 *.eventlog
 .stack-work/
+/hie.yaml
```

Comment:
Could you add a comment here explaining what this file is and why it should be in the .gitignore file?

Reply:
Haskell IDE Engine uses ``hie.yaml`` for project related settings and it should be in the .gitignore file because this file is optional and can vary from person to person.

Reply:
Ah, sorry, I meant edit this PR and literally leave a comment about this in the `.gitignore` file.

Also, please open a PR against https://github.com/github/gitignore/blob/master/Haskell.gitignore, since that should help all Haskellers.  (The gitignore file in pretty-simple was originally copied from there.)

Reply:
Please check.

Reply:
Okay, thanks!

I'll merge this in when a similar PR against https://github.com/github/gitignore/blob/master/Haskell.gitignore is approved and merged in.

(Until then, please just add this to your local gitignore if it is annoying: https://stackoverflow.com/questions/5724455/can-i-make-a-user-specific-gitignore-file/5724499)

Reply:
I'll close this pull request for now, I'm not sure if adding to the gitignore will be right way to go.